src/test/java/com/redhat/devtools/lsp4ij/fixtures/LSPSemanticTokensFileViewProviderFixtureTestCase.java

```diff
@@ -0,0 +1,289 @@
+            false,
+            ThreeState.NO,
+            ThreeState.NO
+    );
+    protected static final Consumer<LSPSemanticTokenPsiElement> isTypeDeclaration = element -> assertSemanticToken(
+            element,
+            (fileViewProvider, offset) -> fileViewProvider.isDeclaration(offset),
+            true,
+            ThreeState.YES,
+            ThreeState.YES
+    );
+    protected static final Consumer<LSPSemanticTokenPsiElement> isNonTypeDeclaration = element -> assertSemanticToken(
+            element,
+            (fileViewProvider, offset) -> fileViewProvider.isDeclaration(offset),
+            true,
+            ThreeState.YES,
+            ThreeState.NO
+    );
+    protected static final Consumer<LSPSemanticTokenPsiElement> isTypeReference = element -> assertSemanticToken(
+            element,
+            (fileViewProvider, offset) -> fileViewProvider.isReference(offset),
+            false,
+            ThreeState.YES,
+            ThreeState.YES
+    );
+    protected static final Consumer<LSPSemanticTokenPsiElement> isNonTypeReference = element -> assertSemanticToken(
+            element,
+            (fileViewProvider, offset) -> fileViewProvider.isReference(offset),
+            false,
+            ThreeState.YES,
+            ThreeState.NO
+    );
+    protected static final Consumer<LSPSemanticTokenPsiElement> isUnknown = element -> assertSemanticToken(
+            element,
+            (fileViewProvider, offset) -> fileViewProvider.isUnknown(offset),
+            false,
+            ThreeState.UNSURE,
+            ThreeState.UNSURE
+    );
+
+    private interface TokenTypeVerifier {
+        boolean test(@NotNull LSPSemanticTokensFileViewProvider fileViewProvider, @NotNull Integer offset);
+    }
+
+    private static void assertSemanticToken(
+            @NotNull LSPSemanticTokenPsiElement element,
+            @NotNull TokenTypeVerifier tokenTypeVerifier,
+            boolean isNameIdentifier,
+            @NotNull ThreeState isIdentifier,
+            @NotNull ThreeState isType
+    ) {
+        LSPSemanticTokensFileViewProvider semanticTokensFileViewProvider = LSPSemanticTokensFileViewProvider.getInstance(element);
+        assertNotNull(semanticTokensFileViewProvider);
+        assertTrue(tokenTypeVerifier.test(semanticTokensFileViewProvider, element.getTextOffset()));
+        int offset = element.getTextOffset();
+        assertEquals(isNameIdentifier, element.getNameIdentifier() != null);
+        assertEquals(isIdentifier, semanticTokensFileViewProvider.isIdentifier(offset));
+        assertEquals(isType, semanticTokensFileViewProvider.isType(offset));
+        assertFalse(semanticTokensFileViewProvider.isWhitespace(offset));
+    }
+
+    protected LSPSemanticTokensFileViewProviderFixtureTestCase(@NotNull String fileNamePattern,
+                                                               @NotNull String languageId) {
+        super(fileNamePattern);
+        setLanguageId(languageId);
```

Comment:
We must always set a language ID or the file type mapping isn't available via the language server definition. Not sure if that's intentional or not, but see `LanguageServersRegistry#registerAssociation()` which does the following for each mapping type:

```java
if (!StringUtils.isEmpty(languageId)) {
    serverDefinition.registerAssociation(language, languageId);
}
```